engine/storage/volume/src/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java

```diff
@@ -2025,4 +2037,27 @@ public VolumeInfo updateHypervisorSnapshotReserveForVolume(DiskOffering diskOffe
 
         return volFactory.getVolume(volumeId);
     }
+
+    /**
+     *  inserts volume,snapshot details
+     * @param volumeId
+     * @param snapsShotid
+     */
+    public void insertVolumeDetailsVO(Long volumeId,Long snapsShotid)
```

Comment:
This method populates the entity object data and use addDetail method of Dao internally. Due to the naming convention it may lead to confusion. However this is addressed as per the comment.